typescriptify/typescriptify.go

```diff
@@ -554,7 +554,7 @@ func (t *TypeScriptify) convertType(depth int, typeOf reflect.Type, customCode m
 			field.Type = field.Type.Elem()
 		}
 		jsonFieldName := t.getJSONFieldName(field, isPtr)
-		if len(jsonFieldName) == 0 || jsonFieldName == "-" {
+		if len(jsonFieldName) == 0 || jsonFieldName == "-" || jsonFieldName == "-?" {
```

Comment:
Change `jsonFieldName == "-"` with `jsonFieldName[0] == "-"` since if `jsonFieldName` is *anything* starting with `-`-  it would produce an invalid typescript class.